internal/util/util.go

```diff
@@ -135,36 +136,56 @@ func GetVolumeSnapshotClassForStorageClass(provisioner string, snapshotClient sn
 }
 
 // GetVolumeSnapshotContentForVolumeSnapshot returns the volumesnapshotcontent object associated with the volumesnapshot
-func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1beta1api.VolumeSnapshot, snapshotClient snapshotter.SnapshotV1beta1Interface, log logrus.FieldLogger) (*snapshotv1beta1api.VolumeSnapshotContent, error) {
+func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1beta1api.VolumeSnapshot, snapshotClient snapshotter.SnapshotV1beta1Interface, log logrus.FieldLogger, shouldWait bool) (*snapshotv1beta1api.VolumeSnapshotContent, error) {
+	if !shouldWait {
+		if volSnap.Status == nil || volSnap.Status.BoundVolumeSnapshotContentName == nil {
+			// volumesnapshot hasn't been reconciled and we're not waiting for it.
+			return nil, nil
+		}
+		vsc, err := snapshotClient.VolumeSnapshotContents().Get(*volSnap.Status.BoundVolumeSnapshotContentName, metav1.GetOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting volume snapshot content from API")
+		}
+		return vsc, nil
+	}
+
+	// We'll wait 10m for the VSC to be reconciled polling every 5s
+	timeout := 10 * time.Minute
```

Comment:
What's our expectation for how long this will actually take? It should be pretty quick, right, since we don't have to wait for the snapshot to be "ready to use"?

I have a feeling we're going to need to make this configurable eventually - maybe just add a TODO for now.

Reply:
yes, this should be pretty quick. Esp, now that we are trying to get the volumesnapshotcontents in the volumesnapshot backup plugin.